src/commands/workflow/get.ts

```diff
@@ -108,30 +120,96 @@ export default class WorkflowGet extends BaseCommand<typeof WorkflowGet> {
      * Workflow steps table
      */
 
-    const steps = workflow.steps.map((step, index) => ({ ...step, index }));
+    const { steps } = this.flattenSteps(workflow.steps);
```

Comment:
I'm going by the screenshot included in the PR (btw thank you for that!) and thinking out loud here from a product point of view — but i'm not too sure about the proposed output for handling if-else steps. Personally I find it pretty challenging to follow, though maybe it won't be as bad as for simple cases involving only one level depth if-else branches/steps?

One other idea I had while looking at the screenshot, is perhaps we always show the top level depth of steps at a time, but with options to go deeper with interactive prompt? Almost like pagination, but choosing to go up or down the tree (and maybe we just display a breadcrumb) 🤔

Another alternative is to ditch the table format and try to draw it in a graph format, but that's more involved both from design/eng perspectives and I'm not sure if it's worth the investment.

I think this one is maybe worth catching up in person with @cjbell and @samseely.

Reply:
> One other idea I had while looking at the screenshot, is perhaps we always show the top level depth of steps at a time, but with options to go deeper with interactive prompt? Almost like pagination, but choosing to go up or down the tree (and maybe we just display a breadcrumb)

Oh, I like this idea a lot. I think it's definitely worth exploring! But, I do wonder if users will expect a view that shows the full tree, rather than requiring paginating thru levels like so.

But also, thinking out loud, how many people are actually using this command? This data is also available in Dashboard and in the `workflow.json` file, both of which provide a richer view of the complex tree. Maybe it is fine to restrict how we display it here since it's not that widely used and probably mostly used for quick checks by a user at certain points.

> Another alternative is to ditch the table format and try to draw it in a graph format, but that's more involved both from design/eng perspectives and I'm not sure if it's worth the investment.

And yea, I too thought of this and quickly decided it wouldn't be worth the effort.

Reply:
> Maybe it is fine to restrict how we display it here since it's not that widely used and probably mostly used for quick checks by a user at certain points.

Ok I have another idea. Not mutually exclusive to the pagination-style up/down interactive prompt idea, but we could also provide an option to open up the corresponding dashboard page with a hotkey? For example, the `workflow get xyz` command will print out the workflow details in terminal, but with an option to hit "O"(pen) to open up their browser to show the workflow detail page. We can also implement the same thing for other display commands.

This way maybe we could get away with restricting the display to the "top level" view of workflows initially, by making it easier to open up the full GUI page. 

And optionally we could also implement the interactive prompt to go up or down in terminal, but that's something we could punt on for now.

Reply:
I also really like that idea!

However, IMO, if we choose between one of them I would actually propose we prioritize level-paging for branching workflows, rather than the open dashboard hotkey. Feels like we should ensure _some way_ to allow full view of steps list for this command in CLI layer.

Reply:
Good point! I think either of those options can work for the if-else GA release, and i'm happy to defer to you on this whether you want to give level-paging a shot or sync with chris and sam first.

Reply:
Given conversation at stand up today re not wanting to invest too much effort into an interactive CLI for `workflow get`, I think we should instead go for the approach you've laid out @thomaswhyyou:

- When a workflow contains one or more if-else steps, the table represents just the first level of steps in the tree
- For if-else steps, we render out the # of branches and # of steps within for the summary
- We render out a message indicating that more steps are not shown in the CLI, and we provide a link to the workflow detail page in Dashboard that the user can open in their browser of choice

Thoughts @cjbell @thomaswhyyou?

Reply:
> For if-else steps, we render out the # of branches and # of steps within for the summary

I think we could potentially keep the branch rows that show the # of steps and conditions on each branch? And just skip showing steps in the branches.

> We render out a message indicating that more steps are not shown in the CLI, and we provide a link to the workflow detail page in Dashboard that the user can open in their browser of choice

That works for me. If we want to make it a little extra nice, then we could also [open](https://github.com/sindresorhus/open) the url with a hotkey, but not required. One other thing I want to mention is, we will probably need to make an extra request to `whoami` endpoint to get the account slug in order to format the url.

But overall as an approach, keeping the terminal display simple at the top level steps for now, and then making it easy for users to view it in dashboard makes sense to me. We can always revisit to enrich the terminal display at a later time.

Reply:
@thomaswhyyou @cjbell I've just pushed up a refactor here. For workflows with one or more if-else steps, we'll display only the first level of steps, including if-else branches. We'll also render below the steps list table a message prompting the user to open the full view in the Knock Dashboard.

Example: 

<img width="1083" alt="image" src="https://github.com/knocklabs/knock-cli/assets/5049830/f866e61c-d146-430a-880e-e20fb1f52434">

Reply:
IMO I'm not sure how useful it is to even show each branch in this view here? Could we instead show just the if/else line and the number of branches. I know that omits some information but it feels like this isn't really the place to display it anyway? I know we're going around the houses a bit in this PR but again my pref is probably to do the least on this screen and just see where we need to make improvements in the future. FWIW as well, I think a link to open from the terminal in the dashboard is broadly applicable outside of if/else steps.